web/models/types/entity_type.ex

```diff
@@ -0,0 +1,27 @@
+defmodule CoursePlanner.Types.EntityType do
+  @moduledoc """
+    This module introduces a custom type for Etco for checking status in the model
+  """
+  @behaviour Ecto.Type
+  def type, do: :string
+
+  @valid_entity_types ['Planned', 'Active', 'Finished', 'Graduated', 'Frozen', 'Deleted']
+
+  def valid_entity_types, do: @valid_entity_types
+
+  def cast(value) do
```

Comment:
the callbacks `cast/1`, `dump/1`, and `load/1` should return `{:ok, term} | :error` according to the docs, right now you are returning `term | :error`